### PR TITLE
monitoring-plugins: 2.2.0 -> 2.3.0

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Upstream update and cleanups.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I haven't tried everything in here yet - we will be doing a test deployment next week.

### CODE REVIEW

File: pkgs/servers/monitoring/plugins/default.nix [hunk 1]
```diff
@@ -1,86 +1,117 @@
...
+  version = "2.3.0";
 
   src = fetchFromGitHub {
-    owner  = "monitoring-plugins";
-    repo   = "monitoring-plugins";
-    rev    = "v${majorVersion}";
-    sha256 = "1pw7i6d2cnb5nxi2lbkwps2qzz04j9zd86fzpv9ka896b4aqrwv1";
+    owner = "monitoring-plugins";
+    repo = "monitoring-plugins";
+    rev = "v" + lib.versions.majorMinor version;
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/plugins/default.nix:57]:
```suggestion
    rev = "v${lib.versions.majorMinor version}";
```

File: pkgs/servers/monitoring/plugins/default.nix [hunk 2]
```diff
@@ -1,86 +1,117 @@
...
+  # TODO: Awful hack. Grrr... this of course only works on NixOS.
   # Anyway the check that configure performs to figure out the ping
   # syntax is totally impure, because it runs an actual ping to
   # localhost (which won't work for ping6 if IPv6 support isn't
   # configured on the build machine).
-  preConfigure= ''
+  postPatch = ''
     substituteInPlace po/Makefile.in.in \
-      --replace /bin/sh ${stdenv.shell}
+      --replace /bin/sh ${runtimeShell}
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/plugins/default.nix:68]: /bin/sh should be available and a symlink to bash.

File: pkgs/servers/monitoring/plugins/default.nix [hunk 3]
```diff
@@ -1,86 +1,117 @@
...
   '';
 
-  meta = {
+  meta = with lib; {
     description = "Official monitoring plugins for Nagios/Icinga/Sensu and others";
-    homepage    = "https://www.monitoring-plugins.org";
-    license     = licenses.gpl2;
-    platforms   = platforms.linux;
+    homepage = "https://www.monitoring-plugins.org";
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/servers/monitoring/plugins/default.nix:113]: gpl2Only or gpl2Plus?

**Comment by peterhoeg** [comment]
> -    rev = "v" + lib.versions.majorMinor version;
> +    rev = "v${lib.versions.majorMinor version}";

Matter of taste. I prefer the former.

>   /bin/sh should be available and a symlink to bash.

Yes, but it's impure, so we might as well fix up path.

>   gpl2Only or gpl2Plus?

Good catch. Turns out it's now gpl3Only.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118716` at 18646d4c run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>check-wmiplus</li>
    <li>monitoring-plugins</li>
  </ul>
</details>

**Pull request merged**